Update autotuner docs (#1128)
redpanda-data/documentation: docs/reference/rpk/rpk-redpanda/rpk-redpanda-tune.mdx

> ---

Reviewer: I think this page gets autogenerated here: https://github.com/redpanda-data/documentation/tree/dev/tools

Although, I'm not sure if it's a manual process to cut the `rpk-commands.mdx` file into separate topics cc @Deflaimun

title: rpk redpanda tune
toc_min_heading_level: 3
toc_max_heading_level: 4
---

<head>
  <meta name="title" content="rpk redpanda tune (autotuner) | Redpanda Docs" />
  <meta name="description" content="rpk redpanda tune (autotuner)" />
</head>

> `rpk redpanda tune`, also referred to as Redpanda's autotuner, optimizes the performance of your production nodes. It automatically detects the hardware configuration of a node and tunes Linux kernel settings to optimize performance.

Reviewer: ```suggestion
`rpk redpanda tune`, also referred to as Redpanda's autotuner, identifies the hardware configuration on your machine and optimizes the Linux kernel to give you the best performance for running Redpanda.
```

:::important

> To start your user journey with the autotuner, Redpanda recommends you first follow the guides for production deployment:

Reviewer: ```suggestion
You should run the autotuner as part of the production deployment workflow. Redpanda recommends you first follow a guide for production deployment:
```

- [Deploy for Production](../../../../deploy/deployment-option/self-hosted/manual/production/production-deployment) 
- [Tune Kubernetes Worker Nodes for Production](../../../../deploy/deployment-option/self-hosted/kubernetes/kubernetes-tune-workers)

While you follow the guides, consult this reference for details about the autotuner.
:::

> ### Usage

Reviewer: I wonder why we don't use level 2 headings.

    </tr>
  </tbody>
</table>

### Command overview

This section provides the fundamentals of using the autotuner.

> :::tip

Reviewer: I'd put this somewhere higher up under 'Usage'. It's essential info.

When running `rpk redpanda tune`, make sure that your current Linux user has root privileges. The autotuner requires privileged access to the Linux kernel settings.
:::

- To enable a predetermined set of tuners for production, run the [rpk redpanda mode production](../../../../reference/rpk/rpk-redpanda/rpk-redpanda-mode) command. This command modifies settings in the local `redpanda.yaml` configuration file. 

    <details>
        <summary>Enabled features in production mode</summary>
    Enabling production mode sets the following features, including a subset of tuners:

    ```yaml
    redpanda:
        developer_mode: false
    rpk:
        tune_network: true
        tune_disk_scheduler: true
        tune_disk_nomerges: true
        tune_disk_write_cache: true
        tune_disk_irq: true
        tune_cpu: true
        tune_aio_events: true
        tune_clocksource: true
        tune_swappiness: true
        tune_ballast_file: true
        overprovisioned: false
    ``` 

>     </details>

Reviewer: This belongs in the reference for `rpk redpanda mode production`

Author: Good point, will duplicate it there

> - To list the available tuners and to see whether they're enabled or supported (and a reason for if they're unsupported), run the [rpk redpanda tune list](../../../../reference/rpk/rpk-redpanda/rpk-redpanda-tune-list) command.

Reviewer: I'd put this in 'Related topics', but also consider renaming it to 'Suggested reading' in all rpk reference docs for consistency with our other topics.

    
    - Each tuner has a YAML key flag for enabling/disabling itself in `redpanda.yaml`. Most are formed by prepending `rpk.tune_` to the name of the tuner listed by [rpk redpanda tune list](../../../../reference/rpk/rpk-redpanda/rpk-redpanda-tune-list). See the [Tuners](#tuners) section for the exact key for a tuner. For an example of enabling a tuner, the key for the `aio_events` tuner is `rpk.tune_aio_events`, and it can be enabled with the following command:

        ```
        rpk redpanda config set rpk.tune_aio_events true
        ```

- To run all available tuners, use the [rpk redpanda tune](../../../../reference/rpk/rpk-redpanda/rpk-redpanda-tune) command for `all`:

    ```
    rpk redpanda tune all
    ```

- To run a specific tuner, use the [rpk redpanda tune](../../../../reference/rpk/rpk-redpanda/rpk-redpanda-tune) command for the tuner:

    ```
    rpk redpanda tune <tuner>
    ```

- To learn more about a tuner, use the [rpk redpanda tune help](../../../../reference/rpk/rpk-redpanda/rpk-redpanda-tune) command for the tuner:

    ```
    rpk redpanda tune help <tuner>
    ```

> ### Tuners

Reviewer: This probably belongs in the `rpk-redpanda-tune-list` reference rather than here.

Author: On one hand, it's better organization to move this to `tune list`, on the other hand, it's essential info about the tuners that requires at least another click. I agree it belongs better under list; what's the best way to get the reader to go there, as a Tip?

    </tr>
  </tbody>
</table>

### Examples

> This section provides fundamental examples of using the autotuner.

Reviewer: ```suggestion
This section provides examples of using the autotuner.
```
